### ISSUE TITLE
StringListMailbox has no unit tests

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3547]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)| @wsanchez reported|
|-|-|
|Trac ID|trac#3547|
|Type|defect|
|Created|2008-11-17 20:13:11Z|
|Branch|https://github.com/twisted/twisted/tree/stringlistmailbox-tests-3547|

In `twisted/mail/maildir.py`, `StringListMailbox` has no unit tests

<details><summary>Searchable metadata</summary>

```
trac-id__3547 3547
type__defect defect
reporter__wsanchez wsanchez
priority__normal normal
milestone__ 
branch__branches_stringlistmailbox_tests_3547 branches/stringlistmailbox-tests-3547
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1226952791000000 1226952791000000
changetime__1227136154000000 1227136154000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [25424]) Branching to 'stringlistmailbox-tests-3547'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

Pretty straightforward.  Build results: http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/stringlistmailbox-tests-3547

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

Please merge after fixing those flakes:
```
twisted/mail/maildir.py:15: 'cStringIO' imported but unused
twisted/mail/maildir.py:31: 'alias' imported but unused
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [25446]) Remove unused imports

refs [#3547](https://github.com/twisted/twisted/issues/3547)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [25447]) Merge stringlistmailbox-tests-3547

Author: exarkun
Reviewer: therve
Fixes: [#3547](https://github.com/twisted/twisted/issues/3547)

Add unit tests for `twisted.mail.maildir.StringListMailbox` and fix the implementation
to support deletion, undeletion, and syncing.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
